static.go

```diff
@@ -51,28 +53,37 @@ func NewStaticResolver(id string, opt StaticResolverOptions) (*StaticResolver, e
 		r.extra = append(r.extra, rr)
 	}
 	r.rcode = opt.RCode
-
+	
+	r.truncate = opt.Truncate
+	
 	return r, nil
 }
 
 // Resolve a DNS query by returning a fixed response.
 func (r *StaticResolver) Resolve(q *dns.Msg, ci ClientInfo) (*dns.Msg, error) {
 	answer := new(dns.Msg)
 	answer.SetReply(q)
-
-	// Update the name of every answer record to match that of the query
-	answer.Answer = make([]dns.RR, 0, len(r.answer))
-	for _, rr := range r.answer {
-		r := dns.Copy(rr)
-		r.Header().Name = qName(q)
-		answer.Answer = append(answer.Answer, r)
+	
+	if r.truncate {
+		logger(r.id, q, ci).Debug("truncated")
+		
+		answer.Truncated = true
+		answer.Rcode = 0
+	} else {
+		logger(r.id, q, ci).Debug("responding")
+		
+		// Update the name of every answer record to match that of the query
+		answer.Answer = make([]dns.RR, 0, len(r.answer))
+		for _, rr := range r.answer {
+			r := dns.Copy(rr)
+			r.Header().Name = qName(q)
+			answer.Answer = append(answer.Answer, r)
+		}
+		answer.Ns = r.ns
+		answer.Extra = r.extra
+		answer.Rcode = r.rcode
```

Comment:
I think it might be best to preserve the existing logic and just adding the truncate flag to it. So just adding

```go
answer.Truncated = r.truncate
```
should do the trick. That way if there are other things like NS records etc, they aren't ignored when truncate is set. That makes it more flexible. As for logging, you could use `logger(r.id, q, ci).WithField("truncated", r.truncated).Debug("responding")` perhaps. Then you don't need the if...else condition

Reply:
I like it, let me see how I can edit this stuff... :-)

Reply:
Changed it, including the logging and removing the if condition.